### PR TITLE
Add "Retrospective" tab to dashboard

### PR DESCRIPTION
**Pull request by glass-w**
## Description
Starts to address point (1) in #108 

## Todos
- [x] create new tab with compounds that contain experimental pIC50 data.
- [x] add master plot with: the relative free energies between pairs of compounds with experimental data available vs. predicted transformation free energies between them (if available).
- [x] check values plotted and reported are correct (as well as methodology).

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by glass-w** [comment]
fixed #123

**Comment by glass-w** [comment]
last commit fixes #123

**Comment by glass-w** [comment]
An example of how this retrsopective tab looks can be found [here](https://fah-public-data-covid19-moonshot-sprints.s3.us-east-2.amazonaws.com/dashboards-staging/sprint-5debug/sprint-5-x12073-monomer-neutral-simulated/retrospective_analysis/index.html)

**Comment by jchodera** [comment]
Looks like there's an `import plotly` somewhere now:
```
  File "/home/server/miniconda/envs/fah-xchem-staging/bin/fah-xchem", line 33, in <module>                                                                                                                                            [82/1981]
    sys.exit(load_entry_point('fah-xchem==0.1.0+64.g54f1a8f.dirty', 'console_scripts', 'fah-xchem')())                                                                                                                                         
  File "/home/server/miniconda/envs/fah-xchem-staging/bin/fah-xchem", line 25, in importlib_load_entry_point           
    return next(matches).load()                                                                                                                                                                                                                
  File "/home/server/miniconda/envs/fah-xchem-staging/lib/python3.7/site-packages/importlib_metadata/__init__.py", line 93, in load                                                                                                            
    module = import_module(match.group('module'))                                                                                                                                                                                              
  File "/home/server/miniconda/envs/fah-xchem-staging/lib/python3.7/importlib/__init__.py", line 127, in import_module                                                                                                                         
    return _bootstrap._gcd_import(name[level:], package, level)                                                                                                                                                                                
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import                                                      
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load                                                                                                                                                                            
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked                                           
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked                                                                                                                                                                            
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module                                              
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed                                                                                                                                                                 
  File "/home/server/miniconda/envs/fah-xchem-staging/lib/python3.7/site-packages/fah_xchem-0.1.0+64.g54f1a8f.dirty-py3.7.egg/fah_xchem/app.py", line 11, in <module>                                                                          
    from .analysis import analyze_compound_series                                                                      
  File "/home/server/miniconda/envs/fah-xchem-staging/lib/python3.7/site-packages/fah_xchem-0.1.0+64.g54f1a8f.dirty-py3.7.egg/fah_xchem/analysis/__init__.py", line 32, in <module>                                                            
    from .plots import generate_plots                                                                                  
  File "/home/server/miniconda/envs/fah-xchem-staging/lib/python3.7/site-packages/fah_xchem-0.1.0+64.g54f1a8f.dirty-py3.7.egg/fah_xchem/analysis/plots.py", line 23, in <module>                                                               
    from arsenic import plotting                                                                                       
  File "/home/server/miniconda/envs/fah-xchem-staging/lib/python3.7/site-packages/arsenic/plotting.py", line 3, in <module>                                                                                                                    
    from arsenic import stats, plotlying                                                                               
  File "/home/server/miniconda/envs/fah-xchem-staging/lib/python3.7/site-packages/arsenic/plotlying.py", line 3, in <module>                                                                                                                   
    import plotly.graph_objects as go                                                                                  
ModuleNotFoundError: No module named 'plotly'  
```

**Comment by glass-w** [comment]
@jchodera I've added the `plotly` dependecies but I think all of the `Arsenic` packages should be handled when running `pip install` [here](https://github.com/choderalab/fah-xchem/blob/retro-tab/environment.yml#L29). All of the required packages are in the `Arsenic` yaml file [here](https://github.com/openforcefield/Arsenic/blob/master/devtools/conda-envs/test_env.yaml) so it should work without having to install plotly etc separately or adding it to the `fah-xchem` yaml.

**Comment by dotsdl** [comment]
Reviewing this now @glass-w! This is currently the branch being used in production, so would like to get this merged before anything else.

**Comment by glass-w** [comment]
@dotsdl ok! Let me know if there is anything you'd like me to explain.

**Review by dotsdl — APPROVED** [review]
Looks great @glass-w! Merging!

File: fah_xchem/analysis/website/__init__.py [hunk 1]
```diff
@@ -193,6 +194,7 @@ def generate_website(
     template_path = os.path.join(os.path.dirname(__file__), "templates")
     template_loader = jinja2.FileSystemLoader(searchpath=template_path)
     environment = jinja2.Environment(loader=template_loader)
+    environment.filters["np"] = np # numpy
```
[the remark this answers is not in the document]
Comment (glass-w) [reply fah_xchem/analysis/website/__init__.py:197]: I believe this was added my @jchodera

File: fah_xchem/analysis/__init__.py [hunk 2]
```diff
@@ -101,45 +143,212 @@ def analyze_transformation(
         solvent_phase=solvent_phase,
     )
 
+
+def calc_exp_ddg_DEPRECATED(
```
[the remark this answers is not in the document]
Comment (glass-w) [reply fah_xchem/analysis/__init__.py:147]: yes, this is safe to remove!

File: fah_xchem/analysis/website/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (glass-w) [reply fah_xchem/analysis/website/__init__.py:197]: From what I can see it isn't used in the Jinja2 templates for page construction. @jchodera is this actually used / needed?

[the remark this answers is not in the document]
Comment (jchodera) [reply fah_xchem/analysis/website/__init__.py:197]: Safe to remove! This was used at some point in development, but no longer.

File: fah_xchem/analysis/__init__.py [hunk 3]
```diff
@@ -101,45 +143,212 @@ def analyze_transformation(
...
+
+    initial_experimental_data = compounds_by_microstate[
+        transformation.initial_microstate.microstate_id
+    ].metadata.experimental_data
+    final_experimental_data = compounds_by_microstate[
+        transformation.final_microstate.microstate_id
+    ].metadata.experimental_data
+
+    if ("pIC50" in initial_experimental_data) and ("pIC50" in final_experimental_data):
+        exp_ddg_ij_err = 0.2  # TODO check this is correct
```
[the remark this answers is not in the document]
Comment (jchodera) [reply fah_xchem/analysis/__init__.py:253]: Ideally, this should be listed in the JSON file. We'd have to extend the schema to include uncertainty estimates.

Perhaps for now:
(1) make a constant up top
(2) create an issue to extend schema to override this default

File: fah_xchem/analysis/__init__.py [hunk 4]
```diff
@@ -101,45 +143,212 @@ def analyze_transformation(
...
                 results_iter,
-                total=len(series.transformations),
+                total=len(available_transformations),
                 description="Computing transformation free energies",
             )
             if result is not None
         ]
 
+    # Reprocess transformation experimental errors to only include most favorable transformation
+    # NOTE: This is a hack, and should be replaced by a more robust method for accounting for racemic mixtures
```
[the remark this answers is not in the document]
Comment (jchodera) [reply fah_xchem/analysis/__init__.py:323]: A better way to handle this (in a separate PR in future) would be to depart from what we are doing here, and instead have
1.  An **absolute retrospective** tab, showing predicted and experimental absolute free energies, which correctly aggregate the predicted free energy for racemic compounds (or compounds with multiple protonation/tautomeric states)
2. A **relative retrospective** tab, where only relative transformations between unambiguous microstates (e.g. between compounds that only possess a single microstate each) are shown

File: fah_xchem/analysis/__init__.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dotsdl) [reply fah_xchem/analysis/__init__.py:253]: Made this a constant up top, and created issue for override via `AnalysisConfig` as #142.

File: fah_xchem/analysis/__init__.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (dotsdl) [reply fah_xchem/analysis/__init__.py:323]: Added as #143. Thanks @jchodera!

**Pull request merged**